core/src/main/java/org/elasticsearch/tribe/TribeService.java

```diff
@@ -494,4 +556,30 @@ private void addNewIndex(ClusterState tribeState, ClusterBlocks.Builder blocks,
             }
         }
     }
+
+    // pkg-private for testing
+    static Map<String, MetaData.Custom> mergeChangedCustomMetaData(Set<String> changedCustomMetaDataTypeSet,
+                                                                   Function<String, List<MetaData.Custom>> customMetaDataByTribeNode) {
+        Map<String, MetaData.Custom> changedCustomMetaDataMap = new HashMap<>(changedCustomMetaDataTypeSet.size());
+        for (String customMetaDataType : changedCustomMetaDataTypeSet) {
+            List<MetaData.Custom> tribeCustomMetaDataList = customMetaDataByTribeNode.apply(customMetaDataType);
+            if (tribeCustomMetaDataList.isEmpty() == false) {
+                final MetaData.Custom mergedCustomMetaData;
+                if (tribeCustomMetaDataList.size() == 1) {
```

Comment:
I don't think you need an if else structure here? you can set the first and iterate over the rest? This can be simplified